src/routes/legacyRoutes.ts

```diff
@@ -92,7 +97,7 @@ export const registerCurrenciesV1Route: Serverlet<
   const { userId } = asUserIdQuery(query)
   const { currencyCodes } = asRegisterCurrenciesBody(json)
 
-  const user = await User.fetch(userId)
+  const user = (await User.fetch(userId)) ?? new User(null, userId)
```

Comment:
Was it a bug in the old code that it would not fallback to creating the `User` object if it didn't exist? I see the previous code did this in a few places but not everywhere you are doing now.

Reply:
If so, should line https://github.com/EdgeApp/edge-push-server/pull/39/files#diff-08c39429f4d1b8541f2714d54006185cc0b4396e2dfa22c6fbf1493047b9d6ecR50

also have the fallback?

Reply:
Right. The old server would 500 if it failed to look up the user, and now it gives a `TypeError`, which translates to a 400 error - still the wrong response. I could have changed that to a 404, but just doing the fallback seemed simpler, cleaner, and more correct. Any place we already have an `if` statement guarding the lookup we can just leave as-is.